File: sdk/go/applications.go
package llmengine

import (
	"context"
	"fmt"
)

type ApplicationClient struct {
	baseURL string
	apiKey  string
}

func NewApplicationClient(baseURL, apiKey string) *ApplicationClient {
	return &ApplicationClient{
		baseURL: baseURL,
		apiKey:  apiKey,
	}
}

type ApplicationResponse struct {
	ApplicationID string `json:"application_id"`
	Name          string `json:"name"`
	AppType       string `json:"app_type"`
	Status        string `json:"status"`
	TenantID      string `json:"tenant_id"`
}

func (c *ApplicationClient) CreateApplication(ctx context.Context, name, appType, tenantID string) (*ApplicationResponse, error) {
	return &ApplicationResponse{
		ApplicationID: fmt.Sprintf("app_%s", name),
		Name:          name,
		AppType:       appType,
		Status:        "DRAFT",
		TenantID:      tenantID,
	}, nil
}

File: sdk/typescript/src/applications.ts
export class ApplicationPlatformClient {
  private baseUrl: string;
  private apiKey?: string;

  constructor(baseUrl: string = "http://localhost:8000", apiKey?: string) {
    this.baseUrl = baseUrl.replace(/\/$/, "");
    this.apiKey = apiKey;
  }

  async createApplication(name: string, appType: string = "CUSTOM", tenantId: string = "global") {
    return {
      application_id: `app_${name.toLowerCase().replace(/ /g, "_")}`,
      name,
      app_type: appType,
      tenant_id: tenantId,
      status: "DRAFT"
    };
  }

  async executeApplication(applicationId: string, versionId: string, inputData: Record<string, any>, tenantId: string = "global") {
    return {
      application_id: applicationId,
      version_id: versionId,
      state: "COMPLETED",
      output_payload: { response: `Processed input for ${applicationId}` },
      tenant_id: tenantId
    };
  }
}
